mtbf_driver/mtbf_apps/contacts/app.py

```diff
@@ -0,0 +1,50 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette.by import By
+from marionette import expected
+from marionette.errors import JavascriptException
+from gaiatest.apps.contacts.app import Contacts
+from marionette import Wait
+from gaiatest.apps.base import Base
+
+
+class MTBF_Contacts(Contacts):
+
+    _edit_form_title_locator = (By.ID, 'contact-form-title')
+    _edit_form_close_locator = (By.ID, 'contact-form-header')
+    _cancel_search_button_locator = (By.ID, 'cancel-search')
+
+    def launch(self):
+        Base.launch(self)
+
+        Wait(self.marionette, ignored_exceptions=JavascriptException).until(
+            lambda m: m.execute_script('return window.wrappedJSObject.Contacts.asyncScriptsLoaded;') is True)
+
+        self.close_existing_edit_form()
+
+        self.clear_search_view()
+
+        Wait(self.marionette).until(expected.element_displayed(
+            Wait(self.marionette).until(expected.element_present(
+                *self._settings_button_locator))))
+
+    def clear_search_view(self):
+        if len(self.marionette.find_elements(*self._cancel_search_button_locator)) != 0:
+            if self.marionette.find_element(*self._cancel_search_button_locator).is_displayed():
+                self.marionette.find_element(*self._cancel_search_button_locator).tap()
+
+    def close_existing_edit_form(self):
+        if len(self.marionette.find_elements(*self._edit_form_title_locator)) != 0:
+            if self.marionette.find_element(*self._edit_form_title_locator).text == "Add contact":
+                self.apps.switch_to_displayed_app()
+                self.marionette.execute_async_script("Accessibility.click(arguments[0].shadowRoot.querySelector('button.action-button'));",
+                                                     [self.marionette.find_element(*self._edit_form_close_locator)],
+                                                     special_powers=True)
+
+    def search_contact(self, strSearchNo):
+        self.marionette.find_element(*(By.CSS_SELECTOR, '#search-start > input:nth-child(1)')).tap()
```

Comment:
It's better to put it on a variable. For example:
_search_bar_locator = (By.CSS_SELECTOR, '#search-start > input:nth-child(1)')
It's better for others to understand it faster.